README and CONTRIBUTING clean-up

chastell opened the pull request:
This cleans up the README:

* moves things relevant to the contributing guide to CONTRIBUTING.md and links it,
* spells ‘reek’ uniformly everywhere,
* drops the teletype `reek` stylisation (I’m open to reverting this one if you’re fond of it),
* switches to a flat gem version badge,
* uses less eye-bleeding `Hash` syntax in `demo.rb`,
* changes RubyDoc.info link to the current canonical form and links to the gem’s docs, which seem to be fresher (regenerated on gem pushes, probably),
* fixes typography, capitalisation and hyphenation,
* adjusts some wording to read better.

mvz commented:
I'm not sure about moving stuff to CONTRIBUTING.md. I'd like to  keep that file short and sweet because GitHub links to it from the issue creation page. We don't want people to TL;DR it or think contributing to Reek is too hard.

mvz commented on README.md:
```diff
@@ -91,11 +91,11 @@ $stdin -- 3 warnings:
 ## Code smells
 
-`reek` currently includes checks for some aspects of [Control Couple](https://github.com/troessner/reek/wiki/Control-Couple), [Data Clump](https://github.com/troessner/reek/wiki/Data-Clump), [Feature Envy](https://github.com/troessner/reek/wiki/Feature-Envy), [Large Class](https://github.com/troessner/reek/wiki/Large-Class), [Long Parameter List](https://github.com/troessner/reek/wiki/Long-Parame ...
+reek currently includes checks for some aspects of [Control Couple](https://github.com/troessner/reek/wiki/Control-Couple), [Data Clump](https://github.com/troessner/reek/wiki/Data-Clump), [Feature Envy](https://github.com/troessner/reek/wiki/Feature-Envy), [Large Class](https://github.com/troessner/reek/wiki/Large-Class), [Long Parameter List](https://github.com/troessner/reek/wiki/Long-Paramete ...
```
Let's take the opportunity to wrap this long line ...

mvz commented on README.md:
```diff
@@ -105,40 +105,42 @@ reek --help
-3. Having a file ending with .reek in your HOME directory
+1. Using the CLI `-c` switch (see [_Command-line interface_](#command-line-interface) above)
+2. Having a file ending with `.reek` either in your current working directory or in a parent directory (more on that later)
+3. Having a file ending with `.reek` in your `$HOME` directory
```
How about just "home directory"?

mvz commented on README.md:
````diff
@@ -155,7 +157,7 @@ def smelly_method foo
 ```
 
-This is further explained [here](https://github.com/troessner/reek/wiki/Smell-Suppression)
+This is further explained [under Smell Suppresion](https://github.com/troessner/reek/wiki/Smell-Suppression).
````
I think it's nicer if 'under' is not part of the link.

troessner commented:
>> I'm not sure about moving stuff to CONTRIBUTING.md. I'd like to keep that file short and sweet because GitHub links to it from the issue creation page. We don't want people to TL;DR it or think contributing to Reek is too hard.

:+1: 

>> drops the teletype reek stylisation (I’m open to reverting this one if you’re fond of it),

I vote for reverting this. "reek" is a name in this case, but it's also a proper english verb, so we should style it properly. Otherwise it gets confusing and sentences get weird...

mvz commented on README.md:
```diff
@@ -105,40 +105,42 @@ reek --help
+All options that go beyond the [Basic Smell Options](https://github.com/troessner/reek/wiki/Basic-Smell-Options) should be documented in the corresponding smell type wiki page, but if you want to get a quick and full overview over all possible configurations you can always check out [the `config/default.reek` file in this repository](config/defaults.reek).
 
-Here's an excerpt of a `reek` configuration file from a commercial project:
+Here’s an excerpt of a reek configuration file from a commercial project:
```
How are you inputting "’" and "–"?

mvz replied:
Digging a little deeper, I think all these single right quote characters should be reverted to apostrophes.

chastell replied:
> How are you inputting "’" and "–"?

I use the [pl(intl) keyboard layout](http://marcinwolinski.pl/keyboard/) – on Linux there seem to be a matching ‘international’ one for every major language.

mvz commented on README.md:
```diff
@@ -105,40 +105,42 @@ reek --help
 The first thing you probably want to check out are the [Basic Smell Options](https://github.com/troessner/reek/wiki/Basic-Smell-Options) which are supported by every smell type.
-Certain smell types offer a configuration that goes beyond that of the basic smell options - for instance [Data Clump](https://github.com/troessner/reek/wiki/Data-Clump).
-All options that go beyond the [Basic Smell Options](https://github.com/troessner/reek/wiki/Basic-Smell-Options) should be documented in the corresponding smell type wiki page but if you want to get a quick and full overview over all possible configurations you can always check out [the default.reek file in this repository](https://github.com/troessner/reek/blob/master/config/defaults.reek).
+Certain smell types offer a configuration that goes beyond that of the basic smell options – for instance [Data Clump](https://github.com/troessner/reek/wiki/Data-Clump).
```
I like a good &ndash; or &mdash;, but perhaps we can just use &ldquo;&hellip; smell options, for instance &hellip; &rdquo;? That avoids having to discuss how we encode dashes in markdown.

mvz commented:
> > drops the teletype reek stylisation (I’m open to reverting this one if you’re fond of it),

> I vote for reverting this. "reek" is a name in this case, but it's also a proper english verb, so we should style it properly. Otherwise it gets confusing and sentences get weird...

I agree. We should set it apart as a name either as `reek` or as Reek.

chastell commented:
Updated as per comments:
* removed additions to `CONTRIBUTING.md`,
* reverted the _Developing `reek` / Contributing_ section,
  * I still feel that the `bundle exec rspec spec/your/file_spec.rb:23`, etc. examples make more sense in the contributing guide (or should be removed altogether, as they seem strangely specific for README,
* wrapped the longest paragraphs,
* stylised `reek` everywhere (FWIW: I’m also ok with spelling it with a capital ‘R’),
* switched all curly apostrophes to straight ones,
  * should we also add double spaces after periods and spell Timo’s name as ‘Roessner’? ;)
* removed dashes,
* addressed the other minor fixes.

troessner commented:
Wait, whats wrong with my name?..:)
I mean, i know, the Umlauts and the sharp s, but is this causing problems somewhere?

chastell commented:
Nothing’s wrong with your name, and there are no problems (that was my point)! I was just making a tongue-in-cheek, slippery-slope argument against straight apostrophes – they’re not harder to type then your name (especially once you configure your keyboard properly). ;)

Seriously though: I _do_ understand the argument of ‘I type apostrophes/quotes/dashes/ellipses often and I don’t want to change my habits just to have proper typography’, so I reverted to straight apostrophes. Maybe 2016 will be the year of proper typography on the destop. ;)

mvz commented:
> I type apostrophes/quotes/dashes/ellipses often and I don’t want to change my habits just to have proper typography

Ha! No, my argument was that a single right quote is not an apostrophe. Mind you, I could still be wrong :smile:.

chastell commented:
> a single right quote is not an apostrophe

Oh but [it is](http://en.wikipedia.org/wiki/Apostrophe#Unicode). :smile:

troessner commented:
Nice clean up, merged!

mvz commented:
> Oh but it is. :smile:

I said I could be wrong :smile:. Feel free to move back to proper apostrophes.
